src/js/common/schemaform/SchemaForm.jsx

```diff
@@ -0,0 +1,125 @@
+import React from 'react';
+import _ from 'lodash/fp';
+import Form from 'react-jsonschema-form';
+
+import { uiSchemaValidate, transformErrors } from './validation';
+import Address from './Address';
+import FieldTemplate from './FieldTemplate';
+import * as reviewWidgets from './review/widgets';
+import ReviewFieldTemplate from './review/ReviewFieldTemplate';
+import widgets from './widgets/index';
+import ObjectField from './ObjectField';
+import ArrayField from './ArrayField';
+import ReviewObjectField from './review/ObjectField';
+import { scrollToFirstError } from '../utils/helpers';
+
+const fields = {
+  ObjectField,
+  ArrayField,
+  address: Address
+};
+
+const reviewFields = {
+  ObjectField: ReviewObjectField,
+  ArrayField,
+  address: ReviewObjectField
+};
+
+/*
+ * Each page uses this component and passes in config. This is where most of the page level
+ * form logic should live.
+ */
+class SchemaForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.validate = this.validate.bind(this);
```

Comment:
I don't mean the syntax for arrow functions is hard to google, I mean the syntax for auto binding class methods used here is (i.e. setting `validate` to equal an arrow function). It's not valid ES2015, it's a Babel [transform](https://babeljs.io/docs/plugins/transform-class-properties/) that's bundled in one of the presets we use.

I don't have any problems with arrow functions; this is a different (though related) feature.